src/main/scala/gov/nih/nlm/ncbi/finagle/consul/catalog/ConsulCatalogResolver.scala

```diff
@@ -66,8 +74,13 @@ class ConsulCatalogResolver extends Resolver {
         val as = jsonToAddresses(parse(response.getContentString()))
         update() = Addr.Bound(as.map(Address(_)))
         val idx = response.headerMap.getOrElse("X-Consul-Index", "0")
+
+        Try(idx.toFloat).foreach(index => consulIndex = index)
+
         cycle(idx)
-      case t: Throw[_] => timer.doLater(Duration(1, TimeUnit.SECONDS)) {
+      case Throw(t) => timer.doLater(Duration(1, TimeUnit.SECONDS)) {
+        log.warning(t, s"Exception throw while querying Consul for service discovery")
```

Comment:
lets also add counter for these exceptions, that will help to correlate issues in graphs